src/wp-includes/link-template.php

```diff
@@ -89,6 +89,56 @@ function permalink_anchor( $mode = 'id' ) {
 	}
 }
 
+/**
+ * Determine whether post should always use an ugly permalink structure.
+ *
+ * @since 5.7.0
+ *
+ * @param WP_Post|int|null $post   Optional. Post ID or post object. Defaults to global $post.
+ * @param bool             $sample Optional. Whether to force consideration based on sample links.
```

Comment:
Maybe add something like `If omitted, relies on the filter status of the passed post`?